lib/dry/configurable/class_methods.rb

```diff
@@ -12,8 +12,16 @@ def inherited(subclass)
         super
 
         subclass.instance_variable_set(:@__config_extension__, __config_extension__)
-        subclass.instance_variable_set(:@_settings, _settings.dup)
-        subclass.instance_variable_set(:@_config, config.dup) if respond_to?(:config)
+
+        new_settings = _settings.dup
+        subclass.instance_variable_set(:@_settings, new_settings)
+
+        # Only classes **extending** Dry::Configurable have class-level config. When
+        # Dry::Configurable is **included**, the class-level config method is undefined because it
+        # resides at the instance-level instead (see `Configurable.included`).
+        if respond_to?(:config)
+          subclass.instance_variable_set(:@config, config.dup_for_settings(new_settings))
```

Comment:
Interestingly, until now we had been setting :@_config, when the actual memoized instance variable (used further below in the `#config` method) is `@config`. Setting this to the memoized ivar is in fact the desired behaviour here, so I think before now things must have been working accidentally.

Reply:
@timriley it'd be good to decide on just one approach for naming private ivars, in hanami I noticed we have `_foo`, in dry-rb we have `__foo__`. I'm just *mentioning* it here for future consideration.